### ISSUE TITLE
[jest-puppeteer] Protocol error (Runtime.callFunctionOn): Target closed

### ISSUE THREAD

**Issue by BoruiGu** [Netflix/pollyjs#208]
## Description

Rep:

- clone jest-puppeteer example https://github.com/Netflix/pollyjs/tree/master/examples/jest-puppeteer
- run `yarn`
- run `yarn test`

### Error Message & Stack Trace

```
yarn run v1.16.0
$ jest
 PASS  __tests__/dummy-app.test.js (8.486s)
  jest-puppeteer
    ✓ should be able to navigate to all routes (7509ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        8.615s
Ran all test suites.
(node:19697) UnhandledPromiseRejectionWarning: Error: Protocol error (Runtime.callFunctionOn): Target closed.
    at Promise (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Connection.js:182:12)
    at ExecutionContext.evaluateHandle (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/ExecutionContext.js:106:44)
    at ExecutionContext.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at ExecutionContext.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/ExecutionContext.js:48:31)
    at ExecutionContext.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at DOMWorld.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/DOMWorld.js:112:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  -- ASYNC --
    at Frame.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:110:27)
    at Page.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Page.js:782:43)
    at Page.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:171:14)
    at new Promise (<anonymous>)
    at new F (.../pollyjs/examples/jest-puppeteer/node_modules/@babel/runtime-corejs2/node_modules/core-js/library/modules/_export.js:36:28)
    at PuppeteerAdapter.passthroughRequest (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:166:30)
    at PuppeteerAdapter.passthroughRequest [as onPassthrough] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:265:33)
    at PuppeteerAdapter.onPassthrough [as onRecord] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:240:16)
    at PuppeteerAdapter.onRecord [as record] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:169:17)
  -- ASYNC --
    at Page.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:110:27)
    at evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:171:14)
    at new Promise (<anonymous>)
    at new F (.../pollyjs/examples/jest-puppeteer/node_modules/@babel/runtime-corejs2/node_modules/core-js/library/modules/_export.js:36:28)
    at PuppeteerAdapter.passthroughRequest (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:166:30)
    at PuppeteerAdapter.passthroughRequest [as onPassthrough] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:265:33)
    at PuppeteerAdapter.onPassthrough [as onRecord] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:240:16)
    at PuppeteerAdapter.onRecord [as record] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:169:17)
    at PuppeteerAdapter.record (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:194:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:19697) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:19697) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:19697) UnhandledPromiseRejectionWarning: Error: Protocol error (Runtime.callFunctionOn): Target closed.
    at Promise (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Connection.js:182:12)
    at ExecutionContext.evaluateHandle (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/ExecutionContext.js:106:44)
    at ExecutionContext.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at ExecutionContext.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/ExecutionContext.js:48:31)
    at ExecutionContext.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at DOMWorld.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/DOMWorld.js:112:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  -- ASYNC --
    at Frame.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:110:27)
    at Page.evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/Page.js:782:43)
    at Page.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:111:23)
    at evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:171:14)
    at new Promise (<anonymous>)
    at new F (.../pollyjs/examples/jest-puppeteer/node_modules/@babel/runtime-corejs2/node_modules/core-js/library/modules/_export.js:36:28)
    at PuppeteerAdapter.passthroughRequest (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:166:30)
    at PuppeteerAdapter.passthroughRequest [as onPassthrough] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:265:33)
    at PuppeteerAdapter.onPassthrough [as onRecord] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:240:16)
    at PuppeteerAdapter.onRecord [as record] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:169:17)
  -- ASYNC --
    at Page.<anonymous> (.../pollyjs/examples/jest-puppeteer/node_modules/puppeteer/lib/helper.js:110:27)
    at evaluate (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:171:14)
    at new Promise (<anonymous>)
    at new F (.../pollyjs/examples/jest-puppeteer/node_modules/@babel/runtime-corejs2/node_modules/core-js/library/modules/_export.js:36:28)
    at PuppeteerAdapter.passthroughRequest (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter-puppeteer/src/index.js:166:30)
    at PuppeteerAdapter.passthroughRequest [as onPassthrough] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:265:33)
    at PuppeteerAdapter.onPassthrough [as onRecord] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:240:16)
    at PuppeteerAdapter.onRecord [as record] (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:169:17)
    at PuppeteerAdapter.record (.../pollyjs/examples/jest-puppeteer/node_modules/@pollyjs/adapter/src/index.js:194:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:19697) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
✨  Done in 15.96s.
```

## Environment

```
Node.js v10.15.1
darwin 18.2.0
6.9.0
1.16.0
```

**Comment by offirgolan** [comment]
@BoruiGu Thank you for bringing this to our attention. I'll investigate this further and see what could be causing this.

**Comment by offirgolan** [comment]
@BoruiGu this should now be resolved with 8ade90d5fe0f.
